src/engine/controls/loopingcontrol.cpp

```diff
@@ -1400,12 +1401,17 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint, bool enable
         return;
     }
 
-    // If resizing an inactive loop by changing beatloop_size,
-    // do not seek to the adjusted loop.
-    newloopInfo.seekMode = (keepStartPoint && (enable || m_bLoopingEnabled))
-            ? LoopSeekMode::Changed
-            : LoopSeekMode::MovedOut;
-
+    // Only update seek mode if the endpoints have changed sufficiently.
+    if (positionNear(newloopInfo.startPosition, loopInfo.startPosition) &&
+            positionNear(newloopInfo.endPosition, loopInfo.endPosition)) {
+        newloopInfo.seekMode = loopInfo.seekMode;
```

Comment:
when you say "new play position" do you mean "new start position"? This function has no concept of "old and new play head position".  

It is fine to activate loops when the play head is before the loop start position, this is the intended behavior.

Reply:
The current play position is relevant when seek mode is used. In a rare condition it can be in the gap between the old and new end position. Let's say loopInfo.endPosition 101, play-position at 100.5 and  newloopInfo.endPosition = 100. 
Without LoopSeekMode::MovedOut; the loop will be exited. I think that is undesired,

Reply:
ok, updated